hooks/taskcreator.js

```diff
@@ -0,0 +1,93 @@
+var assert      = require('assert');
+var slugid      = require('slugid');
+var taskcluster = require('taskcluster-client');
+var debug       = require('debug')('hooks:taskcreator');
+var _           = require('lodash');
+
+class TaskCreator {
+  /** Create a TaskCreator instance.
+   *
+   * options:
+   * {
+   *   credentials:   // TaskCluster credentials
+   * }
+   * */
+  constructor(options) {
+    assert(options, "options must be given");
+    assert(options.credentials instanceof Object,
+        "Expected credentials");
+    this.credentials = options.credentials;
+  }
+
+  taskForHook(hook) {
+    let task = _.cloneDeep(hook.task);
+    task.created = new Date().toJSON();
+    task.deadline = taskcluster.fromNow(hook.deadline).toJSON();
```

Comment:
there is an actual `taskcluster.fromNowJSON()` method too...
Oh, and `taskcluster.fromNowJSON() === new Date().toJSON()`